src/Http/Hal/HalDeserializer.php

```diff
@@ -57,9 +57,13 @@ public static function deserializeResource($entityClass, HalDocument $halDocumen
     public static function deserializeResources($entityClass, string $rel, HalDocument $halDocument): HalResources
     {
         if ($halDocument->hasEmbedded($rel)) {
+            $embedded = $halDocument->getEmbedded($rel);
+            if (!is_array($embedded)) {
+                $embedded = [$embedded];
+            }
             $resources = array_map(function (HalDocument $embeddedDocument) use ($entityClass) {
                 return self::deserializeResource($entityClass, $embeddedDocument);
-            }, $halDocument->getEmbedded($rel));
+            }, $embedded);
```

Comment:
Is it possible this fails because `$halDocument->hasEmbedded($rel)` always returns `true` now (after the API change), but line 64 exploded because the empty array is de-serialized in an unexpected way?